### PR TITLE
python test fixes

### PR DESCRIPTION
**Pull request by jjsjann123**
fix python kernel arg check segfault:
    1. disabled rank check because of active fusion requirement.
    2. put Fusion inside cudaKernel to facilitate runtime arg check.
add shape propagation for newly added opeartion: [addcmul, lerp]

### CODE REVIEW

File: test/cpp/jit/test_gpu.cpp [hunk 1]
```diff
@@ -776,15 +776,15 @@ void testGPU_FusionParser() {
     }
   }
 
-  Fusion fusion;
-  FusionGuard fg(&fusion);
   torch::jit::fuser::cuda::CudaKernel prog;
+  Fusion& fusion = *prog.fusion_;
+  FusionGuard fg(&fusion);
```
Comment (naoyam) [inline test/cpp/jit/test_gpu.cpp:848]: Does it make sense to have a shortcut like below?
```
torch::jit::fuser::cuda::CudaKernel prog;
FusionGuard fg(prog);
```

**Comment by naoyam** [comment]
LGTM. Just added a minor comment.

**Review by naoyam — APPROVED** [review]

File: test/cpp/jit/test_gpu.cpp (hunk 1 above)
Comment (jjsjann123) [reply test/cpp/jit/test_gpu.cpp:848]: I put a wrapper to construct `FusionGuard` from `CudaKernel` and it is used in kernel.cpp.
But here `fusion` is also used in the `GPULower` down and I think it doesn't hurt to leave fusion here.

**Comment by jjsjann123** [comment]
tests are fixed now. I'm merging this one.

**Pull request merged**